topcoffea/modules/HistEFT.py

```diff
@@ -53,10 +53,35 @@ def _init_sumw2(self):
       else:
         self._sumw2[key] = self._sumw[key].copy()
             
-  def set_wilson_coefficients(self,values):
-    """Set the WC values used to evaluate the bin contents of this histogram"""
+  def set_wilson_coeff_from_array(self, values):
+    """Set the WC values used to evaluate the bin contents of this histogram to the contents of 
+       an array where the elements are ordered according to the order defined by wcnames.
+    """
     self._wcs = np.asarray(values).copy()
 
+  def set_sm(self):
+    """Conveniece method to set the WC values to SM (all zero)"""
+    self._wcs = np.zeros(self._nwc)
+    
+  def set_wilson_coefficients(self, **values):
+    """Set the WC values used to evaluate the bin contents of this histogram
+       where the WCs are specified as keyword arguments.  Any WCs not listed are set to zero.
+    """
+    self.set_sm()
+    for key in values:
+      try:
+        index = self._wcnames.index(key)
```

Comment:
Have you profiled this? `index()` can be slow, but with the small number of WCs we have, it might not be a big deal. The other option is use `enumerate()` to build a dictionary of `name: index` values.

Reply:
I have not profiled, but given that we're not expecting our wc list to grow much bigger than maybe 30 or so, I'm not worried.  The enumerate +  dictionary option will take some time to build the dictionary and then look up the keys in it, so for a small list, I'm not sure you'll see a big difference.

Reply:
OK, I timed setting the WC values using `index` and was getting numbers around 4μs for WC lists in the 16 to 26 length range.  Since you'd need to call this > 250 times to add a second on to a piece of code, that seems fast enough to me.  (Setting the WC values 250 times in one piece of code shouldn't happen all that often.  Maybe just in the data-card maker when the templates are getting made.)  Of course, if you're setting a lot of WC values, you can also use the `set_wilson_coeff_from_array()` method to avoid the index call.  Anyway, let me know if you think the performance of this function will cause a bottleneck and I can work on trying to optimize it.